### PR TITLE
Run tests on first boot

### PR DESCRIPTION
**Pull request by gabriel-v**
This PR sets up the `wait_first_boot.py` script to run two test suites:
- the liquid-core tests
- a new test suite for setup

### CODE REVIEW

File: ansible/roles/common/templates/boot.sh [hunk 1]
```diff
@@ -13,19 +13,24 @@ echo 'supervisor up'
...
+
 # Errors on, try to start first boot.
 set -e
 cd /opt/common
 if [ ! -f first_boot_done ] && [ ! -f first_boot_failed ]; then
   echo "Starting first boot."
   if ./initialize.sh ; then
     echo "First boot done."
-    touch first_boot_done
+    #touch first_boot_done
```
Comment (mgax) [inline ansible/roles/common/templates/boot.sh]: Ok, you're touching the file at the end, so just remove this line.

File: ansible/roles/common/templates/boot.sh [hunk 2]
```diff
@@ -43,4 +48,27 @@ do
   "$file"
 done
 
+# Mark first_boot_done and first_boot_failed only after the
```
Comment (mgax) [inline ansible/roles/common/templates/boot.sh]: Meh, no need to leave the comment here. Move it to the commit message, so the intrepid code archeologist can find it, one year from now, when we've rewritten the file 3 times.

File: bin/wait_first_boot.py [hunk 3]
```diff
@@ -34,6 +36,57 @@ def cat_log(message, log_filename=FILE_LOG):
...
+            env = os.environ.copy()
+            env.update(self.env)
+        else:
+            env = None
+        for command in self.pre_commands:
+            print("+", command)
+            subprocess.run([command], shell=True, check=True, env=env)
+        pytest_cmd = [self.pytest, '--junit-xml', self.xml_file]
+        print("+", " ".join(pytest_cmd))
+        subprocess.run(pytest_cmd, cwd=self.chdir, env=env)
```
Comment (mgax) [inline bin/wait_first_boot.py]: Please add `check=False` so the intention is clear.

File: test/.gitignore [hunk 4]
```diff
@@ -0,0 +1,2 @@
+/venv
```
Comment (mgax) [inline test/.gitignore]: Maybe name the whole folder `tests`? The plural form feels more appropriate.

File: ansible/roles/common/templates/boot.sh (hunk 2 above)
Comment (gabriel-v) [reply ansible/roles/common/templates/boot.sh]: I'm leaving the comment there for us to it to python later without breaking the tests in a non-reproducible way (say, some services are not started when `wait_first_boot` picks up the file).

When someone refactors that spaghetti, they'll kill the comments too.

Comment (mgax) [reply ansible/roles/common/templates/boot.sh]: :+1:

**Pull request merged**